src/main/java/io/jenkins/docker/connector/DockerComputerSSHConnector.java

```diff
@@ -253,8 +254,12 @@ protected ComputerLauncher createLauncher(DockerAPI api, String workdir, Inspect
         if ( retryWaitTimeOrNull!=null ) {
             connectionCheck.withEveryRetryWaitFor( retryWaitTimeOrNull, TimeUnit.SECONDS );
         }
+        final Integer launchTimeoutSeconds = getLaunchTimeoutSeconds();
+        if( launchTimeoutSeconds != null) {
+            connectionCheckSSH = connectionCheckSSH.withSSHTimeout(launchTimeoutSeconds, TimeUnit.SECONDS);
```

Comment:
While I'm in favor of _something_ calling `withSSHTimeout`, the `launchTimeoutSeconds` value was never intended to be used for this purpose.
It would be wrong for the code to use the SSHLauncher's `launchTimeoutSeconds` field to _also_ be the SSH timeout value as that's not as-documented and not as-intended; it would cause confusion if we reuse this field without documenting it as such, and it's going to be difficult to explain WTF the field is if we try to document it (and, in my experience, when code's behavior is difficult to justify in the documention, that means the code is wrong!).
Unfortunately there is no other existing configuration field that is suitable here, so we _either_ have to add a new configuration field explicitly for this (which we could do, but would be more work), or we just leave it with its default value (which hasn't been a problem in the last N years, so it's unlikely to be a big issue now).
i.e. don't use this field, either add a new one _or_ just don't set it at all.

...alternatively:  Promise me that, after the ssh-slaves-plugin introduces a new field for the readTimeout, you submit another PR for this plugin that adds a field of the same name as SSHLauncher's readTimeout field that pass the field to the SSHLauncher when that's create _and_ _also_ use it at this point in the code for the `withSSHTimeout` value :grinning:

Reply:
>While I'm in favor of something calling withSSHTimeout, the launchTimeoutSeconds value was never intended to be used for this purpose.

It is the value passed from UI and used everywhere, this value is changed when you change`Connection Timeout in Seconds` setting in the UI, so I guess it is the correct value

<img width="1084" alt="Screenshot 2019-10-15 at 09 32 10" src="https://user-images.githubusercontent.com/5400788/66811825-0dc61780-ef32-11e9-95b4-1f8bfff543ae.png">

https://github.com/jenkinsci/docker-plugin/blob/master/src/main/java/com/nirima/jenkins/plugins/docker/launcher/DockerComputerSSHLauncher.java#L30
https://github.com/jenkinsci/docker-plugin/blob/master/src/main/java/io/jenkins/docker/connector/DockerComputerSSHConnector.java#L359-L362

Reply:
:+1:  I agree that the `launchTimeoutSeconds` field is the one described as `Connection Timeout in Seconds` in the UI and that the configuration logic for all of that is working as expected.
:-1:  What I _don't_ agree with is using that field for the `withSSHTimeout` value.

The "Connection Timeout in Seconds"/`launchTimeoutSeconds` field (in the DockerComputerSSHLauncher) is equivalent to (and used as) the field of the same name in the SSHLauncher class.  That field is documented (by the ssh-slaves-plugin) as:
> Set the timeout value for ssh agent launch in seconds. If empty, it will be reset to default value.
> This will only set the timeout for agent launching; once launched, the timeout will not apply.

i.e. the `launchTimeoutSeconds` field holds the timeout that the SSHLauncher uses to set a limit on the time it takes for the already-logged-in SSH connection to result in an active running slave.jar process before the SSHLauncher declares a failure; it is _not_ the time limit for the SSH daemon to respond to a login, so it'd be wrong to use it as that.

The SSHLauncher (currently) has no "timeout while waiting for progress during the SSH login process" field (and this is why the DockerComputerSSHLauncher has no equivalent).  The `launchTimeoutSeconds` field (that it does have) is *not* intended for (or described as) that purpose, so we should not use it for that purpose.

TL;DR: Our `withSSHTimeout` method sets a _different kind_ of connection timeout to the one described by the "Connection Timeout in Seconds" field.

Reply:
>i.e. the launchTimeoutSeconds field holds the timeout that the SSHLauncher uses to set a limit on the time it takes for the already-logged-in SSH connection to result in an active running slave.jar process before the SSHLauncher declares a failure; it is not the time limit for the SSH daemon to respond to a login, so it'd be wrong to use it as that.

in the ssh-slaves-plugin, `launchTimeoutSeconds` is used as timeout for the SSH connection, the "Connection Timeout in Seconds" in the ssh-slaves-plugin is the same field than "Connection Timeout in Seconds" in the DockerComputerSSHLauncher, in this PR the `launchTimeoutSeconds` is used as timeout for the SSH connect (same ssh-slaves), read, and connection(same ssh-slaves). `ConnectionCheckSSH` checks if it is possible to connect to the SSH agent, for that uses trilead-ssh2 connect method with the same parameter it would use for the Launcher, I do not see any need to define a `timeout while waiting for progress during the SSH login process` because `ConnectionCheckSSH` is checking the connection and it uses the parameter defined for the SSHLauncher, for me it is correct, by adding another parameter to the docker-plugin only for the SSH would be overhead more the UI, making difficult to tune the plugin even more.